doc/glossary.md

```diff
@@ -0,0 +1,13 @@
+# Glossary
+
+This document is intended to name and describe the concepts, functions and data structures inside Kaleidoscope.
+
+It is, as yet, incredibly incomplete.
+
+Entries should be included in dictionary order. When describing an identifier of any kind from the codebase, it should be 
+written using identical capitalization to its use in the code and surrounded by backticks: `identifierName`
+
+
+Event handler
+
+: A function, usually provided by a `Plugin` that is run by a `Hook`
```

Comment:
__Plugin__ is not well defined because it is used for the class `Plugin` and also for Arduino libraries, that sometimes don't contain `Plugin` classes. Might be a good candidate for the glossary.